tests/test_data_utils/test_select_values.py

```diff
@@ -20,21 +20,44 @@ def setUp(self):
             obs={'column1': ['X', 'Y', 'X', 'Y', 'X', 'Z']}
         )
 
+    def test_dataframe_nonexistent_annotation(self):
+        """
+        Test error raised for a nonexistent annotation in a DataFrame.
+        """
+        with self.assertRaises(KeyError):
+            select_values(self.df, 'nonexistent_column', ['A', 'B'])
+
+    def test_adata_nonexistent_annotation(self):
+        """
+        Test error raised for a nonexistent annotation in an AnnData object.
+        """
+        with self.assertRaises(KeyError):
+            select_values(self.adata, 'nonexistent_column', ['X', 'Y'])
+
     def test_select_values_dataframe_typical_case(self):
         """
         Test selecting specified values from a DataFrame column.
         """
         result_df = select_values(self.df, 'column1', ['A', 'B'])
         # Expecting 5 rows where column1 is either 'A' or 'B'
         self.assertEqual(len(result_df), 5)
+        # Check that the 'column1' only contains the values 'A' and 'B'
+        unique_values_in_result = result_df['column1'].unique().tolist()
+        self.assertTrue(set(unique_values_in_result).issubset(set(['A', 'B'])))
+        # Alternatively, assert directly the expected values are in the result
+        expected_values = ['A', 'B']
+        for value in unique_values_in_result:
+            self.assertIn(value, expected_values)
```

Comment:
@fangliu117 , the assert should make sure the two sets are identical. Would you use something like assertCountEqual instead.